sphinx/environment/collectors/title.py

```diff
@@ -50,7 +51,7 @@ def process_doc(self, app: Sphinx, doctree: nodes.document) -> None:
             break
         else:
             # document has no title
-            titlenode += nodes.Text('<no title>')
+            titlenode += explicit_title or nodes.Text('<no title>')
```

Comment:
I think new variable is not needed for this change:
```suggestion
            titlenode += nodes.Text(document.get('title') or '<no title>')
```